### ISSUE TITLE
Conflict upgrading project to path when other dependency depends on path.py

### ISSUE THREAD

**Issue by SethMMorton** [jaraco/path#195]
Consider the following case, when `foo` has `path.py` as a dependency.

```
% pip install --find-links=./pypi foo "path>=14"
Looking in links: pypi
Processing ./pypi/foo-3.0.1-py2.py3-none-any.whl
Collecting path>=14
  Downloading path-14.0.1-py3-none-any.whl (21 kB)
Collecting path.py>=11.2
  Using cached path.py-12.4.0-py3-none-any.whl (2.3 kB)
Collecting importlib-metadata>=0.5; python_version < "3.8"
  Using cached importlib_metadata-1.6.0-py2.py3-none-any.whl (30 kB)
Collecting zipp>=0.5
  Using cached zipp-3.1.0-py3-none-any.whl (4.9 kB)
ERROR: path-py 12.4.0 has requirement path<13.2, but you'll have path 14.0.1 which is incompatible.
Installing collected packages: zipp, importlib-metadata, path, path.py, foo
Successfully installed importlib-metadata-1.6.0 path-14.0.1 path.py-12.4.0 foo-3.0.1 zipp-3.1.0
```

In my project, I'd like to update to the newest `path` to remove deprecation errors when calling `lines()`, but I cannot because my project *also* depends on `foo`, which lists `path.py` and not `path` as a dep.

In the above example, things work out because `pip` does not actually offer conflict management. However, in my actual case I am using `pip-compile` to truely resolve conflicts, and for that it fails:

```
% cat requirements.in                                  
foo
path>=14
% pip-compile --find-links=./pypi
Could not find a version that matches path<13.2,>=14 (from -r requirements.in (line 2))
Tried: 13.0.0, 13.0.0, 13.1.0, 13.1.0, 13.2.0, 13.2.0, 13.3.0, 13.3.0, 14.0.0, 14.0.0, 14.0.1, 14.0.1
There are incompatible versions in the resolved dependencies:
  path>=14 (from -r requirements.in (line 2))
  path<13.2 (from path.py==12.4.0->foo==3.0.1->-r requirements.in (line 1))
```

The same would happen to users of `pipenv` or `poetry`. 

Would you be willing/interested in releasing one more major version of `path.py` that just depends on `path` without specifying a version? I realize you probably specified an upper version to force users to change their `requirements.txt` to point to `path` instead, and I definitely want to do that long term. But this conflict prevents me from doing this in the step-wise manner in which I would prefer to approach the problem.

**Comment by SethMMorton** [comment]
Just checking in to see if this is a possibility or not. I was hoping to find out one way or another so that I know if I need to orchestrate a stepwise move from `path.py` to `path` or if I will be able to do it smoothly.

**Comment by jaraco** [comment]
> Would you be willing/interested in releasing one more major version of `path.py` that just depends on `path` without specifying a version?

I have no specific objection. I only wonder - are there any downsides? I can't think of any. I guess we'll find out.

**Comment by jaraco** [comment]
Fixed in 12.5.0.
